main/settings/src/main/scala/sbt/Reference.scala

```diff
@@ -78,7 +78,7 @@ object Reference {
       case LocalRootProject    => "{<this>}<root>"
       case LocalProject(id)    => "{<this>}" + id
       case RootProject(uri)    => "{" + uri + " }<root>"
-      case ProjectRef(uri, id) => "{" + uri + "}" + id
+      case ProjectRef(uri, id) => s"""ProjectRef(uri("$uri"),"$id")"""
```

Comment:
I really don't see why we have to do this. We're doing something that is useless in most of the occasions. Can you please explain a common use case for which copy-pasteability is desirable for project refs? Project references are used only in exceptional cases where direct access to project definitions is not possible.
    
If we're going to optimize for a use case, we should optimize for the project definition name. This `root / Compile / compile / scalacOptions` instead of `ProjectRef(uri("file:/xxx/helloworld/"),"root")/Compile/scalacOptions`.